pennylane/ops/op_math/prod.py

```diff
@@ -314,3 +320,44 @@ def queue(self, context=qml.QueuingContext):
             context.safe_update_info(op, owner=self)
         context.append(self, owns=self.factors)
         return self
+
+    @property
+    def arithmetic_depth(self) -> int:
+        return 1 + max(factor.arithmetic_depth for factor in self.factors)
+
+    @classmethod
+    def _simplify_factors(cls, factors: Tuple[Operator]) -> Tuple[Operator]:
+        """Reduces the depth of nested factors.
+
+        Returns:
+            Tuple[List[~.operation.Operator], List[~.operation.Operator]: reduced sum and non-sum
+            factors
+        """
+        new_factors = ()
+
+        for factor in factors:
+            if isinstance(factor, Prod):
+                tmp_factors = cls._simplify_factors(factors=factor.factors)
+                new_factors += tmp_factors
+                continue
+            simplified_factor = factor.simplify()
+            if isinstance(simplified_factor, Prod):
+                new_factors += (simplified_factor.factors,)
+            elif isinstance(simplified_factor, Sum):
+                new_factors += (simplified_factor.summands,)
```

Comment:
It seems problematic to me that these two cases are treated the same? Also codecov complaining about missing test coverage.

Reply:
Would the following two input factors yield the same output? 

- `factors = ( X, Sum(Y, Z) )`. ---> `new_factors = (X, Y, Z)`
- `factors = ( X, Adjoint(Prod(Z, Y)) )` ---> `new_factors = (X, Y, Z)`

Reply:
I just changed this XD. They have different behaviour now.